server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -374,11 +374,14 @@ protected static Collection<ResolvedExpression> resolveExpressionsToResources(Co
                 return List.of();
             }
         } else {
-            Predicate<String> isMatchAll = (((Predicate<String>) Metadata.ALL::equals)).or(Regex::isMatchAllPattern);
             if (expressions == null
                 || expressions.length == 0
                 || expressions.length == 1
-                    && (SelectorResolver.selectorsValidatedAndMatchesPredicate(expressions[0], context, isMatchAll))) {
+                    && (SelectorResolver.selectorsValidatedAndMatchesPredicate(
+                        expressions[0],
+                        context,
+                        s -> Metadata.ALL.equals(s) || Regex.isMatchAllPattern(s)
+                    ))) {
```

Comment:
Out of curiosity, hat's the value of inlining this variable?

Reply:
Metadata.all::equals and the `.or` call both actually allocate an object so this removes allocation outright (as well as some overhead for the method handles).
Also, the `.or` doesn't compile anything really, it just chains two predicates generically, unless that inlines we're just burning extra cycles for method calls vs a non-capturing lambda :)